### DISCUSSION TITLE
urql + pinia + hrm

### DISCUSSION THREAD

**Discussion by fnzr** [urql-graphql/urql#2878] [Q&A] [answered]
Hey, I'm setting up an environment to use urql inside a pinia store.
Normal usage works beautifully, but I'm having trouble with HMR for the store.
The gist of it is:
store.js
```javascript
export const useSampleStore = defineStore('sample', () => {
    const index = ref(0)
    const {data} = useQuery({
        query: gql`query($id: ID!) { person(id: $id){ name }}`,
        variables: computed(() => ({id: ids[index.value]}))
    })
    const name = computed(() => data.value?.person.name || "not found")
    return {index, name}
})

if (import.meta.hot) {
    import.meta.hot.accept(acceptHMRUpdate(useSampleStore, import.meta.hot))
}
```
App.vue
```vue
<script setup>
import {useSampleStore} from "@/sampleStore.js";
const store = useSampleStore()
</script>
<template>
  <main>
    <button @click="store.index++">Next id</button>
      <h2>{{ store.name }}</h2>
  </main>
</template>
```

This works exactly as expected: clicking the button changes the store state and a new query is executed as the computed variable changes.

The problem happens on the hot module replacement: on hot reload, urql crashes with the message "Uncaught (in promise) Error: use* functions may only be called during the `setup()` or other lifecycle hooks" and a full reload is necessary.

I dont think this is strictly a urql bug since it's related to its interaction with pinia HMR, but it's it's the only thing preventing the entire thing from just working.

Can I work around this problem somehow and make urql play nice with pinia's HMR?

**Comment by JoviDeCroock** [comment, marked as answer]
Hmm, that would mean `Vue` is returning no `currentInstance` after the HMR invocation 😅 https://github.com/urql-graphql/urql/blob/main/packages/vue-urql/src/useClient.ts#L27-L31

EDIT: oh, `use*` should not be used within a pinia store as it isn't a real component which risks not unsubscribing 😅 related: https://github.com/urql-graphql/urql/discussions/2422#discussioncomment-3999680
